scripts/taginfo.js

```diff
@@ -2,22 +2,57 @@
 
 import * as fs from "fs";
 import * as ShieldDef from "../src/js/shield_defs.js";
+import * as Shields from "../src/js/shield.js";
+import * as Gfx from "../src/js/screen_gfx.js";
+import * as CustomShields from "../src/js/custom_shields.js";
+import * as skia from "skia-canvas";
+
+if (!fs.existsSync("dist/shield-sample")) {
+  fs.mkdirSync("dist/shield-sample", true);
+}
+
+//Headless graphics context
+Gfx.setGfxFactory((bounds) => {
+  let canvas = new skia.Canvas(bounds.width, bounds.height);
+  let ctx = canvas.getContext("2d");
+  ctx.imageSmoothingQuality = "high";
+  ctx.textAlign = "center";
+  ctx.textBaseline = "top";
+  ctx.canvas.width = bounds.width;
+  ctx.canvas.height = bounds.height;
+  return ctx;
+});
+
+//Hash function to elminiate duplicate shield graphics
+/*
+    Released into public comain by bryc (github.com/bryc)
+    Source: https://github.com/bryc/code/blob/master/jshash/experimental/cyrb53.js
+*/
```

Comment:
Yes, it would produce JSON text, but you could strip out all the non-alphanumeric characters using a regular expression. Even better, take those raw stringified JSON strings and use them as keys in a dictionary, mapping them to unique, incrementing integers; use those integers as the actual file names.

Reply:
Updated to use incrementing integers as suggested.